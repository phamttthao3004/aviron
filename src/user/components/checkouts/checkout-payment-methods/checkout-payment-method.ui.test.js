import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import CONSTANTS from '../../../scripts/apps/constants';
import CheckoutPaymentMethodUI from './checkout-payment-method.ui';

describe('<CheckoutPaymentMethodUI />', () => {
    it('h1 contains correct text', () => {
        const props = { clickElement: jest.fn(), goBack: jest.fn() };
        const wrapper = shallow(<CheckoutPaymentMethodUI {...props} />);
        expect(wrapper.find('.title').text()).toBe('Payment Method');
    });
    it('Render Checkout UI component matches the snapshot', () => {
        const props = { clickElement: jest.fn(), goBack: jest.fn() };
        const tree = shallow(<CheckoutPaymentMethodUI {...props} />);
        expect(shallowToJson(tree)).toMatchSnapshot();
    });

    it('Button click go to next step', () => {
        const clickElement = jest.fn();
        const goBack = jest.fn();
        const wrapper = shallow(<CheckoutPaymentMethodUI clickElement={clickElement} goBack={goBack} />);
        const history = { push: jest.fn() };
        const pathname = { pathname: CONSTANTS.NAVIGATION_URL.CHECKOUT_REVIEW_ORDER };
        wrapper.find('.btn-pay').simulate('click');
        expect(clickElement).toHaveBeenCalled(history.push(pathname));
    });
    it('Button click go back', () => {
        const clickElement = jest.fn();
        const goBack = jest.fn();
        const wrapper = shallow(<CheckoutPaymentMethodUI clickElement={clickElement} goBack={goBack} />);
        const history = { push: jest.fn() };
        const pathname = { pathname: CONSTANTS.NAVIGATION_URL.CHECKOUT_INFORMATION };
        wrapper.find('.btn-goback').simulate('click');
        expect(goBack).toHaveBeenCalled(history.push(pathname));
    });
});
