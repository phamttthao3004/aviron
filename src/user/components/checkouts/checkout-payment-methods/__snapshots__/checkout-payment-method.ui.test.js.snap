// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckoutPaymentMethodUI /> Render Checkout UI component matches the snapshot 1`] = `
<div
  className="checkout-content"
>
  <div
    className="title"
  >
    Payment Method
  </div>
  <div
    className="row"
  >
    <div
      className="col-sm-4 col-xs-12"
    >
      <p>
        <b>
          Credit Card
        </b>
      </p>
    </div>
    <div
      className="col-md-8 col-xs-12 text-right"
    >
      <p>
        <img
          alt=""
          src="/images/user/payment-card.png"
        />
      </p>
    </div>
  </div>
  <br />
  <div
    className="row"
  >
    <div
      className="col-md-6 col-xs-12"
    >
      <p>
        Credit Card Number
      </p>
      <div
        className="input-group"
      >
        <input
          className="form-control"
          placeholder=""
          type="text"
        />
      </div>
    </div>
    <div
      className="col-md-3 col-xs-6"
    >
      <p>
        Exp (MM/YY)
      </p>
      <div
        className="input-group"
      >
        <input
          className="form-control"
          placeholder=""
          type="text"
        />
      </div>
    </div>
    <div
      className="col-md-3 col-xs-6"
    >
      <p>
        CCV/CVC
      </p>
      <div
        className="input-group"
      >
        <input
          aria-describedby="basic-addon2"
          className="form-control"
          placeholder=""
          type="text"
        />
        <span
          className="input-group-addon"
          id="basic-addon2"
        >
          <i
            className="icon-c-question"
          />
        </span>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-xs-12"
    >
      <p>
        Cardholder Name
      </p>
      <div
        className="input-group"
      >
        <input
          className="form-control"
          placeholder=""
          type="text"
        />
      </div>
    </div>
  </div>
  <div
    className="checkbox"
  >
    <label
      htmlFor="confirm"
    >
      <input
        name="confirm"
        type="checkbox"
        value=""
      />
      Billing address same as shipping
    </label>
  </div>
  <br />
  <div
    className="row"
  >
    <div
      className="col-xs-6"
    >
      <button
        className="btn btn-goback"
        onClick={[MockFunction]}
        type="button"
      >
        Back
      </button>
    </div>
    <div
      className="col-xs-6 text-right"
    >
      <button
        className="btn btn-pay"
        onClick={[MockFunction]}
        type="button"
      >
        Review order
      </button>
    </div>
  </div>
</div>
`;
